### PR TITLE
Document metadata only downloads

### LINKED ISSUE #308: Add more metadata documentation
While working with some ScPCA metadata, I noticed that we have some undocumented columns in the `single_cell_metadata.tsv` files, such as `unfiltered_cells`, `filtered_cells` (for multiplex), and `processed_cells`. While the column names are intuitive, it's still best not to have to guess, however educated!

 Metadata changes are going to be hitting the portal at some near-future time, and we should go through and make sure that all metadata columns are clearly documented.

### LINKED ISSUE #309: Add sample metadata file to Downloadable files section
Add a subsection under `Downloadable Files` section explaining the sample metadata file and its contents. This is for the project level metadata file.

### LINKED ISSUE #310: Add all samples metadata file to Downloadable files section
Add a subsection explaining the all samples metadata file and its contents under `Downloadable Files`. This is for samples metadata from all samples in the portal.

### PR DESCRIPTION
**Pull request by allyhawkins**
Closes #310 
Closes #309 
Closes #308 

Here I am adding documentation for the upcoming metadata only downloads that will be available. 

- I included a few sentences in the first section that mentions metadata only downloads are available and links to the metadata section for more information. 
- I updated the metadata section to include a full table with all expected columns that will be present in all metadata files and in the correct order. See https://github.com/AlexsLemonade/scpca-portal/issues/699#issuecomment-2140625806 for the full order of columns that are expected. 
- I did not include `workflow`, `workflow_version`, and `workflow_commit` in the table since those seemed self explanatory? Let me know if you think I should add them. 
- I added a new section describing the metadata only downloads. Here I just mentioned that metadata can be downloaded separately by project or for the whole Portal. For the contents of the actual metadata, these will mirror what's in the table for the individual metadata with or without the addition of additional project columns. 

Let me know if we think this is enough to cover this addition? I'm definitely open to ideas about organization here since I wasn't totally confident in my placement. I'm curious what others think. 

Just a small note that we had an outdated description of `transcript_type`, so I adjusted that here too.

### CODE REVIEW

File: docs/download_files.md [hunk 1]
```diff
@@ -90,41 +93,77 @@ Therefore, there will be no cell type report in the download for these libraries
 
 ## Metadata
 
-The `single_cell_metadata.tsv` file is a tab-separated table with one row per library and the following columns.
+Included with each download, is a `single_cell_metadata.tsv` file containing relevant metadata for each sample included in the download.
```
Comment (sjspielman) [inline docs/download_files.md:96]:
```suggestion
Included with each download is a `single_cell_metadata.tsv` file containing relevant metadata for each sample included in the download.
```

File: docs/download_files.md [hunk 2]
```diff
@@ -90,41 +93,77 @@ Therefore, there will be no cell type report in the download for these libraries
...
 Examples of this include treatment or outcome.
-Metadata pertaining to processing will also be available in this table and inside of the `SingleCellExperiment` object.
+Metadata pertaining to processing will be available in this table and inside of the `SingleCellExperiment` and `AnnData` objects.
 See the {ref}`SingleCellExperiment experiment metadata <sce_file_contents:singlecellexperiment experiment metadata>` section for more information on metadata columns that can be found in the `SingleCellExperiment` object.
 See the {ref}`AnnData experiment metadata <sce_file_contents:anndata experiment metadata>` section for more information on metadata columns that can be found in the `AnnData` object.
 
 For projects with bulk RNA-seq data, the `bulk_metadata.tsv` file will be included for project downloads.
 This file will contain fields equivalent to those found in the `single_cell_metadata.tsv` related to processing the sample, but will not contain patient or disease specific metadata (e.g. `age`, `sex`, `diagnosis`, `subdiagnosis`, `tissue_location`, or `disease_timing`).
 
+### Metadata only downloads
```
Comment (sjspielman) [inline docs/download_files.md:158]: I think? (and check any relative links, too)
```suggestion
### Metadata-only downloads
```

File: docs/download_files.md [hunk 3]
```diff
@@ -90,41 +93,77 @@ Therefore, there will be no cell type report in the download for these libraries
...
+
+Metadata for all samples on the Portal is available to download separately from gene expression data downloads.
+The metadata for all samples in a given project can be downloaded as a single zip file containing the `metadata.tsv` file and a `README.md` file.
```
Comment (sjspielman) [inline docs/download_files.md:161]: What do you think of this?
```suggestion
Each project page has an option to download metadata for all of its samples as a single zip file containing the `metadata.tsv` file and a `README.md` file.
```

File: docs/download_files.md [hunk 4]
```diff
@@ -90,41 +93,77 @@ Therefore, there will be no cell type report in the download for these libraries
...
+Project-specific metadata will contain all columns mentioned [above table](#metadata) and any additional project-specific columns, such as treatment or outcome.
```
Comment (sjspielman) [inline docs/download_files.md:162]:
```suggestion
Project-specific metadata will contain all columns listed in [the above table](#metadata) and any additional project-specific columns, such as treatment or outcome.
```

File: docs/download_files.md [hunk 5]
```diff
@@ -90,41 +93,77 @@ Therefore, there will be no cell type report in the download for these libraries
...
+
+Additionally, the metadata for all samples on the Portal is available and contains the metadata for all samples available on the Portal.
+The Portal-wide metadata will contain all columns mentioned in the [above table](#metadata).
```
Comment (sjspielman) [inline docs/download_files.md:165]:
```suggestion
The Portal-wide metadata will contain all columns listed in [the above table](#metadata).
```

**Review by sjspielman — COMMENTED** [review]
This seems fine to me overall!

I also wonder how this relates to this other issue I had opened - #308. I think it actually mostly takes care of it, but looks like `filtered_cells` doesn't appear in the table still, unless I'm missing something?

**Comment by allyhawkins** [comment]
> I also wonder how this relates to this other issue I had opened - https://github.com/AlexsLemonade/scpca-docs/issues/308. I think it actually mostly takes care of it, but looks like filtered_cells doesn't appear in the table still, unless I'm missing something?

This should close that as `filtered_cells` is not a column in the metadata that is on the Portal? At least when I download a sample it's not there and it's not included in the list of columns we are using in https://github.com/AlexsLemonade/scpca-portal/issues/699#issuecomment-2140625806. It is `filtered_cell_count`.

**Comment by sjspielman** [comment]
> This should close that as filtered_cells is not a column in the metadata that is on the Portal

It is, but only for _multiplexed_ libraries. Per https://github.com/AlexsLemonade/scpca-portal/issues/699#issuecomment-2140625806, we're not providing this variable in the overall metadata, but it is present in the multiplexed TSVs (just confirmed with a download from SCPCP000009). If it's not supposed to be there either, that's a different problem then!

**Comment by allyhawkins** [comment]
> > This should close that as filtered_cells is not a column in the metadata that is on the Portal
> 
> It is, but only for _multiplexed_ libraries. Per [AlexsLemonade/scpca-portal#699 (comment)](https://github.com/AlexsLemonade/scpca-portal/issues/699#issuecomment-2140625806), we're not providing this variable in the overall metadata, but it is present in the multiplexed TSVs (just confirmed with a download from SCPCP000009). If it's not supposed to be there either, that's a different problem then!

Okay, I think we have that because we don't want to show `filtered_cell_count` on the front end for multiplexed since it's per sample. So I added a note about that column here and then also included `demux_samples` since I had missed that.

**Review by sjspielman — APPROVED** [review]
LGTM! Close em all out!

**Pull request merged**